website/orders/api/v1/urls.py

```diff
@@ -0,0 +1,36 @@
+from django.urls import path, register_converter
+
+from orders.api.v1.views import (
+    OrderView,
+    ShiftListCreateAPIView,
+    shift_add_time,
+    shift_add_capacity,
+    shift_scanner,
+    OrderListAPIView,
+    OrderRetrieveUpdateDestroyAPIView,
+    ProductListAPIView,
+    ShiftRetrieveUpdateAPIView,
+    product_search,
+    OrderCreateAPIView,
+)
+from orders.converters import ShiftConverter
+
+register_converter(ShiftConverter, "shift")
+
+urlpatterns = [
+    path("shifts", ShiftListCreateAPIView.as_view(), name="shifts_listcreate"),
+    path("shifts/<int:pk>", ShiftRetrieveUpdateAPIView.as_view(), name="shift_retrieveupdate"),
+    path("shifts/<shift:shift>/orders", OrderListAPIView.as_view(), name="orders_list"),
+    path(
+        "shifts/<shift:shift>/orders/<int:pk>",
+        OrderRetrieveUpdateDestroyAPIView.as_view(),
+        name="orders_retrieveupdatedestroy",
+    ),
+    path("shifts/<shift:shift>/orders/add", OrderCreateAPIView.as_view(), name="orders_create"),
```

Comment:
If you put the OrderCreateAPIView and the OrderListAPIView together you can properly implement the creation of orders as a POST request. Decide what you do based on the HTTP method and not the route.

Reply:
I did it this way because the OrderRetrieveUpdateDestroy view and the OrderCreate view use different serializers because of the way we create Order objects in the backend. It seemed more natural to create a different API endpoint for both views as they do not rely on the same set of data that they let the API user access or create.